cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetContext.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset;
+
+import co.cask.cdap.api.annotation.Beta;
+
+/**
+ * Provides access to the context for a dataset including its environment and configuration
```

Comment:
I know we discussed it and you guys decided to go the other way, but it still looks confusing to mix at least two things here:
* dataset instance identification (namespace, name - identifies dataset instance)
* environment/context where the dataset object is being instantiated and used

Oh, and per javadoc, dataset configuration is added to the mix,,,